File: SWEA/D2/1288. 새로운 불면증 치료법/새로운 불면증 치료법.py
T = int(input())

for t in range(1, T + 1):
    N = int(input())

    visited = set()  # 본 숫자 저장
    k = 1

    while True:
        num = N * k  # 배수

        # 자리수 확인
        for digit in str(num):
            visited.add(digit)

        # 0~9 다 모였으면 종료
        if len(visited) == 10:
            print(f"#{t} {num}")
            break

        k += 1

File: SWEA/D2/1288. 새로운 불면증 치료법/새로운 불면증 치료법.java
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int T = sc.nextInt(); // 테스트 케이스 수

        for (int t = 1; t <= T; t++) {
            int N = sc.nextInt(); // 입력

            boolean[] visited = new boolean[10]; // 0~9 체크 배열
            int count = 0; // 몇 개 숫자 봤는지
            int k = 1; // 배수

            while (true) {
                int num = N * k; // k번째 배수

                // 자리수 분해
                int temp = num;
                while (temp > 0) {
                    int digit = temp % 10; // 마지막 자리
                    if (!visited[digit]) {
                        visited[digit] = true;
                        count++; // 새로운 숫자 발견
                    }
                    temp /= 10;
                }

                // 0~9 모두 확인했으면 종료
                if (count == 10) {
                    System.out.println("#" + t + " " + num);
                    break;
                }

                k++; // 다음 배수
            }
        }
    }
}